File: 2023-2024/sourcecode/mod3/RetrofitBasicsJava/app/src/main/java/com/example/retrofitbasicsjava/ui/PostFragment.java
package com.example.retrofitbasicsjava.ui;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.retrofitbasicsjava.R;
import com.example.retrofitbasicsjava.databinding.FragmentPostBinding;
import com.example.retrofitbasicsjava.viewmodel.PostViewModel;

public class PostFragment extends Fragment {
    private FragmentPostBinding binding;
    private PostViewModel viewModel;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentPostBinding.inflate(inflater, container, false);
        View rootView = binding.getRoot();

        viewModel = new ViewModelProvider(this).get(PostViewModel.class);

        PostAdapter postAdapter = new PostAdapter(new PostComparator());

        viewModel.posts.observe(getViewLifecycleOwner(), postAdapter::submitList);

        binding.recycler.setLayoutManager(new LinearLayoutManager(requireContext()));
        binding.recycler.setAdapter(postAdapter);

        return rootView;
    }
}

File: 2023-2024/sourcecode/mod3/RetrofitBasicsKotlin/app/src/main/java/com/example/retrofitbasicskotlin/ui/PostFragment.kt
package com.example.retrofitbasicskotlin.ui

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.retrofitbasicskotlin.R
import com.example.retrofitbasicskotlin.databinding.FragmentPostBinding
import com.example.retrofitbasicskotlin.viewmodel.PostViewModel
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

class PostFragment : Fragment() {

    private lateinit var binding: FragmentPostBinding

    private val viewModel: PostViewModel by viewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentPostBinding.inflate(inflater)

        val userAdapter = PostAdapter(PostComparator())
        viewLifecycleOwner.lifecycleScope.launch {
            viewModel.posts.collectLatest { posts ->
                userAdapter.submitList(posts)
            }
        }

        binding.recycler.apply{
            adapter = userAdapter
            layoutManager = LinearLayoutManager(requireContext())
        }

        return binding.root
    }
}